tools/c7n_azure/c7n_azure/template_utils.py

```diff
@@ -45,6 +49,16 @@ def deploy_resource_template(self, group_name, template_file_name, template_para
         return self.client.deployments.create_or_update(
             group_name, group_name, deployment_properties)
 
+    def resource_exist(self, group_name, resource_name):
+        if not self.client.resource_groups.check_existence(group_name):
+            return False
+
+        r_filter = ("name eq '%s'" % resource_name)
+
+        for resource in self.client.resources.list_by_resource_group(group_name, filter=r_filter):
```

Comment:
Yeah so its a totally naive look at "Yeah it looks like this already exists".  I think it can be much smarter, or we can have a "force redeploy" boolean on provision too and leave this dumb - but it seems to optimize the happy path for now.